### PR TITLE
add function: contains

### PR DESCRIPTION
**Pull request by xlqian**
add generic `contains`

### CODE REVIEW

File: functions.h [hunk 1]
```diff
@@ -173,5 +173,21 @@ void clean_up_weak_ptr(std::vector<boost::weak_ptr<T>>& container) {
 std::string make_adapted_uri_fast(const std::string& ref_uri, size_t s);
 std::string make_adapted_uri(const std::string& ref_uri);
 
+template<int N>
```
Comment (pbougue) [inline functions.h]: A little comment here on the purpose of decl_tag is welcome :)

File: functions.h [hunk 2]
```diff
@@ -173,5 +173,21 @@ void clean_up_weak_ptr(std::vector<boost::weak_ptr<T>>& container) {
...
+struct dec_tag: dec_tag<N-1>{};
+template<>
+struct dec_tag<0>{};
+
+// Hint: the use of decltype and comma operator is for the purpose of SFINAE
+template<class C, class T>
+inline auto contains_impl(const C& c, const T& x, dec_tag<1>) -> decltype(c.find(x), true) { return c.find(x) != std::end(c); }
```
Comment (pbougue) [inline functions.h]: line length :disappointed:

File: functions.h (hunk 1 above)
Comment (xlqian) [inline functions.h]: Ok!

File: functions.h (hunk 2 above)
Comment (xlqian) [inline functions.h]: OK

**Comment by xlqian** [comment]
a second version:

http://ideone.com/bhGK61

**Pull request closed without merging**